src/shared/lib/context/cart-context/cart-context.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState, type ReactElement } from "react";
+import { TotalLineItemQuantityContext } from ".";
+import { getCart } from "../../..";
+
+export function TotalQuantityContextProvider({ children }): ReactElement {
+  const [totalLineItemQuantity, setTotalLineItemQuantity] = useState(0);
+  const [productsCheckout, setProductCheckout] = useState<string[]>([]);
+  const [isDownloadPage, setIsDownloadPage] = useState<boolean>(false);
+
+  useEffect(() => {
+    const CheckProductInCart = async (): Promise<void> => {
+      const productIdInCart: string[] = [];
+      if (isDownloadPage) {
+        const cart = await getCart();
+        const productInCart = cart.lineItems;
+        productInCart.map((product) => {
+          productIdInCart.push(product.productId);
+        });
+        setProductCheckout(productIdInCart);
+        setIsDownloadPage(false);
+      }
+    };
```

Comment:
Любимый map для перебора. Понимаю. 😉 
Тут действительно можно его применить, только немного в другом ключе, чтобы сразу присвоить в нужный массив. Ну и функцию с маленькой буквы.

```suggestion
    const checkProductInCart = async (): Promise<void> => {
      let productIdInCart: string[] = [];
      if (isDownloadPage) {
        const cart = await getCart();
        productIdInCart = cart.lineItems.map((product) => product.productId);
      }
      setProductCheckout(productIdInCart);
      setIsDownloadPage(false);
    };
```